actions/setup/action.yml

```diff
@@ -69,7 +59,6 @@ runs:
           /home/runner/.cache/go-build
           /home/runner/go/pkg/mod
           /home/runner/go/bin
-        key: ${{ steps.cache-key.outputs.cache_key }}
+        key: ${{ runner.os }}-${{ github.ref_name }}-${{ github.workflow }}-${{ github.job }}-${{ inputs.cacheKey }}-${{ inputs.go-version }}-${{ hashFiles('**/go.sum') }}
         restore-keys: |
-          ${{ steps.cache-key.outputs.restore_key }}
-
+          ${{ runner.os }}-${{ github.base_ref }}-${{ github.workflow }}-${{ github.job }}-${{ inputs.cacheKey }}-${{ inputs.go-version }}-${{ hashFiles('**/go.sum') }}
```

Comment:
There are two commits in this PR which changes the cache key behavior.

This "simplify cache keys" commit of this PR simplifies how cache keys are generated a bit. This has the side-effect of always creating a new cache in each PR. I'd be happy to walk you through this all in person because I appreciate this might be difficult to follow efficiently.